### PR TITLE
literaturesuggest: normalize journal title names

### LINKED ISSUE #1822: literaturesuggest: transform journal titles to normalize form
In Invenio 2.x, the journal titles in the form were being transformed to a normalized form using a KB that could be found in the config variable `DEPOSIT_INSPIRE_JOURNALS_KB`.

Now that we don't have Knowledge bases, the transformation could be done using the Journal collection (using ElasticSearch)

### PR DESCRIPTION
**Pull request by jmartinm**
* Using the Journal database, normalize names in publication information.
  (closes #1822)

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

File: inspirehep/modules/records/mappings/records/journals.json [hunk 1]
```diff
@@ -182,7 +182,13 @@
...
                                 "journalautocomplete"
                             ],
-                            "type": "string"
+                            "type": "string",
+                            "fields": {
+                              "raw": {
+                                "type": "string",
+                                "index": "not_analyzed"
+                              }
+                            }
```
Comment (jacquerie) [inline inspirehep/modules/records/mappings/records/journals.json]: Why do you need to amend this mapping? You're not searching here. Or am I missing something?

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 2]
```diff
@@ -208,14 +209,17 @@ def formdata_to_model(obj, formdata):
...
+    if journal_title:
+        title_translation = JournalsSearch().query(
+            'match', title_variants__title__raw=journal_title
+        ).execute()
+        if title_translation:
+            try:
+                data["publication_info"][0]["journal_title"] = \
+                    title_translation[0].short_titles[0].title
+            except (AttributeError, IndexError):
+                pass
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/tasks.py]: Needs tests. I'd mock `execute` and write unit tests for this alongside the others in https://github.com/inspirehep/inspire-next/blob/b7f8c916ae3786d26533dd23c2868e0bd6244a11/tests/unit/literaturesuggest/test_literaturesuggest_tasks.py.

File: inspirehep/modules/records/mappings/records/journals.json [hunk 3]
```diff
@@ -192,6 +198,19 @@
...
                     "type": "string"
+                },
+                "title_variants": {
+                    "properties": {
+                      "title": {
+                        "type": "string",
+                        "fields": {
+                          "raw": {
+                            "type": "string",
+                            "analyzer": "lowercase_analyzer"
```
Comment (jacquerie) [inline inspirehep/modules/records/mappings/records/journals.json]: I would expect that a field called `raw` is `not_analyzed`.

File: inspirehep/modules/literaturesuggest/tasks.py [hunk 4]
```diff
@@ -208,14 +209,17 @@ def formdata_to_model(obj, formdata):
...
-        # journals_kb = dict([(x['key'].lower(), x['value'])
-        #                     for x in get_kb_mappings(current_app.config.get("DEPOSIT_INSPIRE_JOURNALS_KB"))])
-
-        # data['publication_info']['journal_title'] = journals_kb.get(data['publication_info']['journal_title'].lower(),
-        #                                                                 data['publication_info']['journal_title'])
-        # TODO convert using journal records
-        pass
+    journal_title = get_value(data, "publication_info[0].journal_title")
+    if journal_title:
+        title_translation = JournalsSearch().query(
```
Comment (jacquerie) [inline inspirehep/modules/literaturesuggest/tasks.py]: I guess it's `title_variants`.

File: inspirehep/modules/records/mappings/records/journals.json (hunk 1 above)
Comment (jmartinm) [inline inspirehep/modules/records/mappings/records/journals.json]: Right, seems like a leftover, should not be needed.

File: tests/unit/literaturesuggest/test_literaturesuggest_tasks.py [hunk 5]
```diff
@@ -65,15 +98,16 @@ def test_formdata_to_model_populates_schema(u, ui):
     obj = StubObj(data, extra_data)
 
     expected = 'https://localhost:5000/schemas/records/hep.json'
-    result = formdata_to_model(obj, formdata)
+    with app.app_context():
+        result = formdata_to_model(obj, formdata)
```
Comment (jacquerie) [inline tests/unit/literaturesuggest/test_literaturesuggest_tasks.py]: That's the thing: if you required the `app` fixture you already have an application context, so you can just
```python
result = formdata_to_model(obj, formdata)
```

Comment (jmartinm) [inline tests/unit/literaturesuggest/test_literaturesuggest_tasks.py]: ah right. Will amend that.

File: tests/unit/literaturesuggest/test_literaturesuggest_tasks.py [hunk 6]
```diff
@@ -50,9 +53,39 @@ def __init__(self, email):
...
+            "timed_out": False,
+            "took": 123
+        }
+    )
+
+
 @mock.patch('inspirehep.modules.literaturesuggest.tasks.User')
 @mock.patch('inspirehep.modules.literaturesuggest.tasks.UserIdentity')
-def test_formdata_to_model_populates_schema(u, ui):
+def test_formdata_to_model_populates_schema(u, ui, app):
```
Comment (jacquerie) [inline tests/unit/literaturesuggest/test_literaturesuggest_tasks.py]: There's a better solution for this: readd `autouse=True` to the `app` fixture, and remove all current explicit uses of it by using `current_app` instead. This requires a little more work, but will avoid this problem from now on.

**Pull request closed without merging**